pkgs/build-support/fetch7z/default.nix

```diff
@@ -0,0 +1,86 @@
+# This function downloads and unpacks an archive file, such as a zip
+# or tar file. This is primarily useful for dynamically generated
+# archives, such as GitHub's /archive URLs, where the unpacked content
+# of the zip file doesn't change, but the zip file itself may
+# (e.g. due to minor changes in the compression algorithm, or changes
+# in timestamps).
+
+{ lib, fetchurl, p7zip }:
```

Comment:
Might I suggest bsdtar from libarchive instead? bsdtar unpacks quite a lot of file types, including as far as I can tell every format that p7zip supports: https://github.com/libarchive/libarchive#supported-formats

Reply:
Yeah, I am planning to remove all that stuff.

Reply:
I will focus only on the simple stuff and leave the database for a further PR.